pkg/operator/ceohelpers/common.go

```diff
@@ -21,43 +24,34 @@ const MachineDeletionHookName = "EtcdQuorumOperator"
 // MachineDeletionHookOwner holds an owner of the Machine Deletion Hook
 const MachineDeletionHookOwner = "clusteroperator/etcd"
 
-// ReadDesiredControlPlaneReplicasCount simply reds the desired Control Plane replica count from the cluster-config-v1 configmap in the kube-system namespace
-func ReadDesiredControlPlaneReplicasCount(configMapListerForKubeSystemNamespace corev1listers.ConfigMapNamespaceLister) (int, error) {
-	const clusterConfigConfigMapName = "cluster-config-v1"
-	const installConfigKeyName = "install-config"
-
-	clusterConfig, err := configMapListerForKubeSystemNamespace.Get(clusterConfigConfigMapName)
+// ReadDesiredControlPlaneReplicasCount reads the current Control Plane replica count
+func ReadDesiredControlPlaneReplicasCount(operatorClient v1helpers.StaticPodOperatorClient) (int, error) {
+	operatorSpec, _, _, err := operatorClient.GetStaticPodOperatorState()
 	if err != nil {
 		return 0, err
 	}
 
-	rawInstallConfig, hasInstallConfig := clusterConfig.Data[installConfigKeyName]
-	if !hasInstallConfig {
-		return 0, fmt.Errorf("missing required key: %s for cm: %s/kube-system", installConfigKeyName, clusterConfigConfigMapName)
+	unstructuredMergedCfg, err := resourcemerge.MergeProcessConfig(
+		nil,
+		operatorSpec.ObservedConfig.Raw,
+		operatorSpec.UnsupportedConfigOverrides.Raw,
+	)
+	if err != nil {
+		return 0, err
 	}
 
-	var unstructuredInstallConfig map[string]interface{}
-	if err := yaml.Unmarshal([]byte(rawInstallConfig), &unstructuredInstallConfig); err != nil {
-		return 0, fmt.Errorf("failed to unmarshal key: %s to yaml from cm: %s/kube-system, err: %w", installConfigKeyName, clusterConfigConfigMapName, err)
+	var unstructuredConfig map[string]interface{}
```

Comment:
as of today, the etcd config is unstructured, in the future we could make it more structured.